apps/frontend/src/pages/route-details/route-details.tsx

```diff
@@ -34,25 +38,47 @@ import styles from "./styles.module.css";
 
 const RouteDetails = (): React.JSX.Element => {
 	const [isEditMode, setIsEditMode] = useState<boolean>(false);
-	const route = useAppSelector(({ routeDetails }) => routeDetails.route);
-	const user = useAppSelector(({ auth }) => auth.authenticatedUser);
+
+	const { user } = useAppSelector(({ auth }) => ({
+		user: auth.authenticatedUser,
+	}));
+	const { route } = useAppSelector(({ routeDetails }) => ({
+		route: routeDetails.route,
+	}));
+	const { categories } = useAppSelector(({ categories }) => ({
+		categories: categories.categories,
+	}));
+
 	const dataStatus = useAppSelector(
 		({ routeDetails }) => routeDetails.dataStatus,
 	);
 
 	const reviews = useAppSelector(({ routeDetails }) => routeDetails.reviews);
+
 	const isAuthenticatedUser = Boolean(user);
-	const { control, errors, getValues, handleReset } =
+
+	const { control, errors, getValues, handleReset, handleValueSet } =
 		useAppForm<RoutePatchRequestDto>({
 			defaultValues: ROUTE_FORM_DEFAULT_VALUES,
 		});
 	const dispatch = useAppDispatch();
 	const { id: routeId } = useParams<{ id: string }>();
+	const [categoriesOptions, setCategoriesOptions] = useState<
+		{ label: string; value: number }[]
+	>([]);
+	useMemo(() => {
+		const options = categories.map((category) => ({
+			label: category.name,
+			value: category.id,
+		}));
+		setCategoriesOptions(options);
+	}, [categories]);
```

Comment:
the use memo itself should return a computed value, you don't need use state

```suggestion
	const categoriesOptions = useMemo(() => {
		const options = categories.map((category) => ({
			label: category.name,
			value: category.id,
		}));
		return options;
	}, [categories]);
```